### PR TITLE
EOS-26111:S3:Enable log roatation of haproxy with logrotate.d

### PR DESCRIPTION
**Pull request by sachitanands**
Signed-off-by: Sachitanand Shelake <sachitanand.shelake@seagate.com>

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: rpms/s3/s3rpm.spec [hunk 1]
```diff
@@ -334,7 +334,6 @@ echo "[cortx-s3server-rpm] INFO: S3 RPM Clean section completed"
 /opt/seagate/cortx/s3/install/haproxy/haproxy_osver7.cfg
 /opt/seagate/cortx/s3/install/haproxy/haproxy_osver8.cfg
 /opt/seagate/cortx/s3/install/haproxy/logrotate/haproxy
-/opt/seagate/cortx/s3/install/haproxy/logrotate/haproxylogfilerollover.sh
```
Comment (nileshgovande) [inline rpms/s3/s3rpm.spec:337]: Remove this file from source base (git rm)

File: scripts/haproxy/logrotate/haproxy [hunk 2]
```diff
@@ -1,17 +1,16 @@
 /var/log/cortx/haproxy.log
-/var/log/cortx/haproxy-status.log
```
Comment (nileshgovande) [inline scripts/haproxy/logrotate/haproxy:2]: Do all the edits to the file on the fly

File: scripts/provisioning/configcmd.py [hunk 3]
```diff
@@ -281,9 +281,10 @@ def configure_haproxy(self):
 
       # Create main config file for haproxy.
       S3HaproxyConfig(self.url).process()
+      # update the haproxy log rotate config file in /etc/logrotate.d/haproxy
+      self.find_and_replace("/etc/logrotate.d/haproxy", "/var/log/cortx", self.base_log_file_path)
```
Comment (nileshgovande) [inline scripts/provisioning/configcmd.py:285]: This needs to be in 'container init script'

File: rpms/s3/s3rpm.spec (hunk 1 above)
Comment (sachitanands) [reply rpms/s3/s3rpm.spec:337]: done

File: scripts/haproxy/logrotate/haproxy (hunk 2 above)
Comment (sachitanands) [reply scripts/haproxy/logrotate/haproxy:2]: done

**Comment by sachitanands** [comment]
Triggered custom-build : http://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/centos-7.9/job/cortx-all-image-custom-ci/1130/

File: scripts/provisioning/configcmd.py (hunk 3 above)
Comment (sachitanands) [reply scripts/provisioning/configcmd.py]: done. moved this to s3_start file

**Review by nileshgovande — APPROVED** [review]

**Review by vimalraghwani20 — APPROVED** [review]

**Pull request merged**